src/NetworkStructures.jl

```diff
@@ -293,8 +293,14 @@ function construct_mass_matrix(mmv_array, gs)
     else
         mass_matrix = sparse(1.0I,gs.dim_v, gs.dim_v)
         for (i, mm) in enumerate(mmv_array)
-            if mm != I
-                mass_matrix[gs.v_idx[i],gs.v_idx[i]] .= mm
+            if mm != I    
+                if mm isa Array{T, 1} where T
```

Comment:
Should we instead go by the ndims of mm?

```julia
if ndims(mm) = 1
# do one thing
elseif ndims(mm) = 2
# do another thing
else error
```
https://docs.julialang.org/en/v1/base/arrays/#Base.ndims

Reply:
OK, I implemented that